### PR TITLE
fix: updated gh-actions, added no-apl

### PR DESCRIPTION
**Pull request by Ani1357**
This PR:
- adds support for deploying an lke cluster without APL
- uses wildcard domains
- fixes domain name generation 
- add support for deploying with a random(valid) domain if/when needed

### CODE REVIEW

**Review by ferruhcihan — COMMENTED** [review]
LGTM! Just wondering—would it be better to move the `No-apl` option under the `installation profile` instead of `domain zone`?

**Comment by Ani1357** [comment]
> LGTM! Just wondering—would it be better to move the `No-apl` option under the `installation profile` instead of `domain zone`?

I agree. Update code accordingly.

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -43,15 +43,17 @@ on:
         options:
           - minimal
           - minimal-with-team
-          - monitoring-with-team
           - full
           - upgrade
+          - No-Apl
```
Comment (CasLubbers) [inline .github/workflows/integration.yml:48]: Nitpick but lets keep the casing consistent.
```suggestion
          - no-apl
```

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -117,11 +119,14 @@ jobs:
           case "${{ inputs.domain_zone }}" in
             "Zone-1") LINODE_CLUSTER_NAME=${{ github.actor }}-1 ;;
             "Zone-2") LINODE_CLUSTER_NAME=${{ github.actor }}-2 ;;
+            "Random") LINODE_CLUSTER_NAME=${{ github.actor }}-$RANDOM ;;
+            "DNS-Integration") LINODE_CLUSTER_NAME=apl-test-${{ inputs.install_profile }} ;;
           esac
-
+          [[ ${{ inputs.install_profile }} == 'No-Apl' ]] && LINODE_CLUSTER_NAME=$LINODE_CLUSTER_NAME-no-apl
```
Comment (CasLubbers) [inline .github/workflows/integration.yml:125]:
```suggestion
          [[ ${{ inputs.install_profile }} == 'no-apl' ]] && LINODE_CLUSTER_NAME=$LINODE_CLUSTER_NAME-no-apl
```

File: .github/workflows/integration.yml [hunk 3]
```diff
@@ -155,30 +160,33 @@ jobs:
           case "${{ inputs.domain_zone }}" in
             "Zone-1") LINODE_CLUSTER_NAME=${{ github.actor }}-1 ;;
             "Zone-2") LINODE_CLUSTER_NAME=${{ github.actor }}-2 ;;
-            "DNS-Integration") LINODE_CLUSTER_NAME=nightly-apl-test-$RANDOM ;;
+            "Random") LINODE_CLUSTER_NAME=${{ github.actor }}-$RANDOM ;;
+            "DNS-Integration") LINODE_CLUSTER_NAME=apl-test-${{ inputs.install_profile }} ;;
           esac
+          [[ ${{ inputs.install_profile }} == 'No-Apl' ]] && LINODE_CLUSTER_NAME=$LINODE_CLUSTER_NAME-no-apl
```
Comment (CasLubbers) [inline .github/workflows/integration.yml:166]:
```suggestion
          [[ ${{ inputs.install_profile }} == 'no-apl' ]] && LINODE_CLUSTER_NAME=$LINODE_CLUSTER_NAME-no-apl
```

File: .github/workflows/integration.yml [hunk 4]
```diff
@@ -155,30 +160,33 @@ jobs:
...
           echo LINODE_CLUSTER_NAME=$LINODE_CLUSTER_NAME >> $GITHUB_ENV
       - name: Determine exact k8s version
         run: |
           echo LINODE_K8S_VERSION=$(linode-cli lke versions-list --json | jq -ce --arg version "$(echo ${{ matrix.kubernetes_versions }} | sed -E 's/^([0-9]+\.[0-9])$/\10/')" '.[] | select(.id | tostring | startswith($version)) | .id') >> $GITHUB_ENV
-      - name: Creating domain for scheduled integration test
+      - name: Determine domain name to use for scheduled integration test
         env:
           EDGEDNS_ZONE: ${{ secrets.EDGEDNS_ZONE }}
-        if: ${{ inputs.domain_zone == 'DNS-Integration' }}
+        if: ${{ inputs.domain_zone == 'DNS-Integration' && inputs.install_profile != 'No-Apl'}}
```
Comment (CasLubbers) [inline .github/workflows/integration.yml:174]:
```suggestion
        if: ${{ inputs.domain_zone == 'DNS-Integration' && inputs.install_profile != 'no-apl'}}
```

File: .github/workflows/integration.yml [hunk 5]
```diff
@@ -155,30 +160,33 @@ jobs:
...
-          # Generating a random 5 char string
           RAND=$(openssl rand -hex 4)
           DOMAIN="integration-${RAND}.${EDGEDNS_ZONE}"
           echo "::add-mask::$DOMAIN"
           echo DOMAIN=$DOMAIN >> $GITHUB_ENV
-
-      - name: Determine domain name to use
-        if: ${{ inputs.domain_zone != 'DNS-Integration' }}
+      - name: Determine domain name
+        if: ${{ inputs.domain_zone != 'DNS-Integration' && inputs.install_profile != 'No-Apl' }}
```
Comment (CasLubbers) [inline .github/workflows/integration.yml:181]:
```suggestion
        if: ${{ inputs.domain_zone != 'DNS-Integration' && inputs.install_profile != 'no-apl' }}
```

File: .github/workflows/integration.yml [hunk 6]
```diff
@@ -217,6 +225,7 @@ jobs:
             sleep 30
           done
       - name: Save kubectl config with auth token and Get kubectl environment and create docker secret
+        if: ${{ inputs.install_profile != 'No-Apl' }}
```
Comment (CasLubbers) [inline .github/workflows/integration.yml:228]:
```suggestion
        if: ${{ inputs.install_profile != 'no-apl' }}
```

File: .github/workflows/integration.yml [hunk 7]
```diff
@@ -235,14 +244,17 @@ jobs:
           echo "Kubectl context set to linode"
           echo LINODE_CLUSTER_CONTEXT=`kubectl config current-context` >> $GITHUB_ENV
       - name: Create image pull secret on test cluster
+        if: ${{ inputs.install_profile != 'No-Apl' }}
```
Comment (CasLubbers) [inline .github/workflows/integration.yml:247]:
```suggestion
        if: ${{ inputs.install_profile != 'no-apl' }}
```

File: .github/workflows/integration.yml [hunk 8]
```diff
@@ -235,14 +244,17 @@ jobs:
...
         run: |
           kubectl create secret docker-registry reg-otomi-github \
             --docker-server=${{ env.CACHE_REGISTRY }} \
             --docker-username=${{ env.BOT_USERNAME }} \
             --docker-password='${{ secrets.BOT_PULL_TOKEN }}'
       - name: Checkout
+        if: ${{ inputs.install_profile != 'No-Apl' }}
```
Comment (CasLubbers) [inline .github/workflows/integration.yml:254]:
```suggestion
        if: ${{ inputs.install_profile != 'no-apl' }}
```

File: .github/workflows/integration.yml [hunk 9]
```diff
@@ -235,14 +244,17 @@ jobs:
...
         uses: actions/checkout@v4
       - name: Prepare APL chart
+        if: ${{ inputs.install_profile != 'No-Apl' }}
```
Comment (CasLubbers) [inline .github/workflows/integration.yml:257]:
```suggestion
        if: ${{ inputs.install_profile != 'no-apl' }}
```

File: .github/workflows/integration.yml [hunk 10]
```diff
@@ -261,6 +273,7 @@ jobs:
             - reg-otomi-github
           EOF
       - name: APL install
+        if: ${{ inputs.install_profile != 'No-Apl' }}
```
Comment (CasLubbers) [inline .github/workflows/integration.yml:276]:
```suggestion
        if: ${{ inputs.install_profile != 'no-apl' }}
```

**Review by CasLubbers — APPROVED** [review]
Only one remark about inconsistency of No-Apl -> no-apl

**Pull request merged**